lib/ansible/module_utils/basic.py

```diff
@@ -427,6 +426,63 @@ def remove_values(value, no_log_strings):
     return new_value
 
 
+def sanitize_keys(obj, no_log_strings, ignore_keys=None):
+    """ Sanitize the keys in a container object by removing no_log values from key names.
+
+    This is a companion function to the `remove_values()` function.
+
+    :param obj: The container object to sanitize. This must be either a Mapping type object,
+        or a list type object that may or may not have additional Mapping type objects to sanitize.
+    :param no_log_strings: A set of string values we do not want logged.
+    :param ignore_keys: A set of string values of keys to not sanitize.
+
+    :returns: An object with sanitized keys.
+    """
+
+    if ignore_keys is None:
+        ignore_keys = frozenset([])
+
+    no_log_strings = [to_native(s, errors='surrogate_or_strict') for s in no_log_strings]
+
+    # So that we convert no_log_strings only once, the recursive part of this
+    # method is placed within the _sanitize_keys() function.
+    return _sanitize_keys(obj, no_log_strings, ignore_keys)
+
+
+def _sanitize_keys(obj, no_log_strings, ignore_keys):
+    if not isinstance(obj, (Mapping, list)):
+        raise TypeError('Unsupported type for key sanitization.')
+
+    # Return an object of the same type.
+    new_obj = type(obj)()
+
+    # For a list, we just recursively iterate over the items, making recursive
+    # calls for target types.
+    if isinstance(obj, list):
```

Comment:
Throughout this function, I don't think list is correct.  it will miss checking containers which are not lists (for instance, tuples and sets).  The decision tree in `_remove_values_conditions()` might be what you want to model this after.

Reply:
👍